ui/src/app/services/server-config.service.ts

```diff
@@ -0,0 +1,26 @@
+import {Injectable} from '@angular/core';
+import {Observable} from 'rxjs/Observable';
+import {ReplaySubject} from 'rxjs/ReplaySubject';
+
+import {ErrorHandlingService} from 'app/services/error-handling.service';
+
+import {ConfigResponse, ConfigService} from 'generated';
+
+@Injectable()
+export class ServerConfigService {
+  private configObs: Observable<ConfigResponse>;
+  constructor(
+    private configService: ConfigService,
+    private errorHandlingService: ErrorHandlingService
+  ) {}
+
+  public getConfig(): Observable<ConfigResponse> {
+    if (!this.configObs) {
+      // Use of a replaySubject() caches the output of the API call across subscriptions.
+      const subject = new ReplaySubject<ConfigResponse>();
```

Comment:
One last note on this.  The buffer size of a `ReplaySubject` defaults to positive infinity (i.e. cache until memory runs out or the subject is disposed of) if not given; in this case it probably doesn't matter since the whole point is just to cache a single call and never run anything through that subject again, but I thought you'd like to know, particularly since the only way to find that out is to read rxjs's source code 🤷‍♂️

Reply:
Thanks, I did actually see this, infinite was intentional but should always be 1 (but I don't want to mask a bug by explicitly using a 1 buffer).